### ISSUE TITLE
Possible wrong deserialization for quering nested values

### ISSUE THREAD

**Issue by syzer** [florianholzapfel/express-restify-mongoose#51]
1. I *really* like the module.
--
2. It is possible to crash express-restify-mongoose
--
```JavaScript
SyntaxError: Unexpected token i
    at Object.parse (native)
    at buildQuery (\node_modules\express-restify-mongoose\lib\express-restify-mongoose.js:180:30)
    at \node_modules\express-restify-mongoose\lib\express-restify-mongoose.js:485:13
    at contextFilter (\node_modules\express-restify-mongoose\lib\express-restify-mongoose.js:458:13)
    at Object.handle (\node_modules\express-restify-mongoose\lib\express-restify-mongoose.js:484:9)
    at next_layer (\node_modules\express\lib\router\route.js:103:13)
    at Object.cleanQuery [as handle] \node_modules\express-restify-mongoose\lib\express-restify-mongoose.js:113:9)
    at next_layer (\node_modules\express\lib\router\route.js:103:13)
    at Route.dispatch (\node_modules\express\lib\router\route.js:107:5)
    at \node_modules\express\lib\router\index.js:205:24
```
example query:
`curl -X GET -H "Content-Type:application/json" -w "\n" -k -s https://localhost:3000/api/v1/articles/?query={"id":"~666"}`

Would be nice to get a `Bad request`

But I guess someone mentioned this problem in other issue (please correct me if i'm wrong)

3. There is problem parsing requests with non-numeric arrays:
--
`curl -H 'Accept: application/json' -k -w "\n" 'https://localhost:3000/api/v1/articles?id=%5BAnyStringId,767%5D'`
returns OK

`curl -H 'Accept: application/json' -k -w "\n" 'https://localhost:3000/api/v1/articles?id=%5B"AnyStringId",767%5D' `
or
`curl -H 'Accept: application/json' -k -w "\n" 'https://localhost:3000/api/v1/articles?id=\["AnyStringId",767\]' `

return empty array `[]`

Basically any proper `"` around strings will make the query return empty set

I applied hack-around in `/lib/express-restify-mongoose.js` after:
`app.get(uri_items, options.middleware, function(req, res, next) {`
..
to remove trailing `"`

Later I hope I will provide more adequate example.
Ps: the `id` is *not* the `_id`.

4. Custom `_id`
--
Its just me or is impossible to update object with custom `_id`, or create new object with custom id?

5. Also possible to crash the module on wrong queries:
--
`TypeError: Cannot read property 'caster' of undefined`

**Comment by Zertz** [comment]
This does not seem to throw an exception anymore. Invalid JSON is the culprit, marking this as a duplicate of #158
